zktraffic/endpoints/stats_server.py

```diff
@@ -77,27 +79,50 @@ def wakeup(self):
   def has_stats(self):
     return len(self._get_stats('per_path')) > 0
 
-  def _get_stats(self, name, prefix=''):
+  def _get_stats(self, name, prefix='', output_array=False):
     stats_by_opname = self._stats.stats(name, self._max_results)
 
-    stats = {}
+    if output_array:
+        stats_arr = []
+    else:
+        stats = {}
     for opname, opstats in stats_by_opname.items():
       for path, value in opstats.items():
-        stats["%s%s%s" % (prefix, opname, path)] = value
-
-    return stats
+        if output_array:
+            #TODO: split prefix to each own key
+            #TODO: when per_ip: split path by ':' and add IPs to their own key
+            if prefix.endswith('/'):
+                prefix=prefix[:-1]
+            tmp_dict={"opname": opname, "path": "%s%s" % (prefix, path), "value":value}
+            stats_arr.append(tmp_dict)
+        else:
+            stats["%s%s%s" % (prefix, opname, path)] = value
+    if output_array:
+        stats_json=json.dumps(stats_arr)
```

Comment:
What happens if you just return list of dictionaries? It won't get converted into a JSON (a string) automatically to be returned to the client? The original dict was passed as-is, which is why I am curious if there is an issue. Might save you the need to turn it into JSON here if it works.

Reply:
According to https://bottle.readthedocs.io/en/latest/tutorial_app.html#using-bottle-for-a-web-based-todo-list, Bottle seems to automatically transform Python dictionaries into JSON. That's the reason I'm using the json.dumps() here.